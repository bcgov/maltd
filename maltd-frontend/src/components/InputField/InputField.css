.input-field {
  width: 200px;
}
