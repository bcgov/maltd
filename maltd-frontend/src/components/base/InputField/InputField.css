.input-field {
  width: 100%;
}

::-webkit-input-placeholder {
  font-style: italic;
}

:-moz-placeholder {
  font-style: italic;
}

::-moz-placeholder {
  font-style: italic;
}

:-ms-input-placeholder {
  font-style: italic;
}
