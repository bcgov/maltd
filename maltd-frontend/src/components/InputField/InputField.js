import React from 'react';
import { Input } from 'reactstrap';
import PropTypes from 'prop-types';
import './InputField.css';

export default function InputField({ inputField: { name, type, placeholder, valid, invalid, value }, onUpdate }) {
  return (
    <Input
      className="input-field"
      type={type}
      name={name}
      placeholder={placeholder}
      valid={valid}
      invalid={invalid}
      value={value}
      onChange={() => onUpdate()}
    />
  );
}

InputField.propTypes = {
  inputField: PropTypes.shape({
    type: PropTypes.string.isRequired,
    placeholder: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
    valid: PropTypes.bool.isRequired,
    value: PropTypes.string.isRequired
  }),
  onUpdate: PropTypes.func,
};
