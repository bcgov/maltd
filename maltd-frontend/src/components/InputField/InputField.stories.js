/* eslint-disable react/jsx-filename-extension */
import React from 'react';
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import 'bootstrap/dist/css/bootstrap.css';

import InputField from './InputField';

export const inputField = {
  name: 'idir',
  type: 'text',
  placeholder: 'IDIR/username',
  valid: false,
  invalid: false,
  value: '',
};

export const actions = {
  onUpdate: action('onUpdate'),
};

/* eslint-disable react/jsx-props-no-spreading */
storiesOf('InputField', module)
  .add('default', () => <InputField inputField={inputField} {...actions} />)
  .add('valid', () => <InputField inputField={{ ...inputField, valid: true, value: 'IDIR/testuser' }} {...actions} />)
  .add('invalid', () => <InputField inputField={{ ...inputField, invalid: true, value: 'a' }} {...actions} />);
