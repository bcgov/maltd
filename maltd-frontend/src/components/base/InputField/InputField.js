/* eslint-disable react/jsx-props-no-spreading */
import React from "react";
import { Input } from "reactstrap";
import PropTypes from "prop-types";
import "./InputField.css";

export default function InputField({
  inputField: { name, type, placeholder, valid, invalid, value, disabled },
  onChange
}) {
  return (
    <Input
      className="input-field my-2"
      type={type}
      disabled={disabled}
      name={name}
      placeholder={placeholder}
      valid={valid}
      invalid={invalid}
      value={value}
      {...{
        ...(onChange && { onChange })
      }}
    />
  );
}

InputField.propTypes = {
  inputField: PropTypes.shape({
    type: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
    placeholder: PropTypes.string.isRequired,
    valid: PropTypes.bool.isRequired,
    invalid: PropTypes.bool.isRequired,
    value: PropTypes.string.isRequired,
    disabled: PropTypes.bool.isRequired
  }).isRequired,
  onChange: PropTypes.func
};

InputField.defaultProps = {
  onChange: () => {}
};
