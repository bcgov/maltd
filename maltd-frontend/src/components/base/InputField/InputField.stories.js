/* eslint-disable react/jsx-filename-extension */
import React from "react";
import { storiesOf } from "@storybook/react";
import "bootstrap/dist/css/bootstrap.css";
import InputField from "./InputField";

const inputField = {
  name: "idir",
  type: "text",
  placeholder: "IDIR/username",
  valid: false,
  invalid: false,
  value: ""
};

storiesOf("InputField", module)
  .add("default", () => <InputField inputField={inputField} />)
  .add("valid", () => (
    <InputField
      inputField={{ ...inputField, valid: true, value: "IDIR/testuser" }}
    />
  ))
  .add("invalid", () => (
    <InputField inputField={{ ...inputField, invalid: true, value: "a" }} />
  ));
