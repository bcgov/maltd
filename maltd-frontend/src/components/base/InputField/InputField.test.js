/* eslint-disable react/jsx-filename-extension */
import React from "react";
import renderer from "react-test-renderer";
import InputField from "./InputField";

describe("Input Field", () => {
  test("Component renders as expected", () => {
    const component = renderer.create(
      <InputField
        inputField={{
          type: "text",
          name: "myinputfield",
          placeholder: "placeholder",
          valid: false,
          invalid: false,
          value: "idir",
          disabled: false
        }}
      />
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
