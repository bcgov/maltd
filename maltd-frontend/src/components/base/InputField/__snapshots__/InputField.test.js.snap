// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Field Component renders as expected 1`] = `
<input
  className="input-field form-control"
  defaultValue="idir"
  name="myinputfield"
  placeholder="placeholder"
  type="text"
/>
`;
